Implement removing proof methods

jksf opened the pull request.

Commit:
Implement removing proof methods

biern commented on test/signhash.ts:
```diff
@@ -142,4 +142,66 @@ contract('SignHash', accounts => {
+  describe('#removeProof', () => {
+    it('should remove proof method', async () => {
+      const signer = accounts[0];
+      const method = 'http';
```
That's a bit unrelated to the PR, but do you plan to have `http` and `https` as a separate methods? I'm not sure how those should be differentiated. I imagined that there should be just a `domain` method

jksf replied:
`https == HTTP over SSL`. When accessing a static file on a server we are using HTTP protocol in the first place. I thought about using SSL certificate as identity proof but I have no idea yet how to integrate it with our identity verification mechanism. 

I would avoid using `domain` method instead of `http` as it would collide with `dns` custom records verification which is another way to proof the domain.

biern commented on test/signhash.ts:
```diff
@@ -142,4 +142,66 @@ contract('SignHash', accounts => {
+
+  describe('#removeProof', () => {
+    it('should remove proof method', async () => {
+      const signer = accounts[0];
```
Most of the tests use just `accounts[0]` as a signer, it could be defined on a higher level to avoid that duplication. Also, I know this should not matter in our case, but deploying the contract from some other account than the signing one might be a good idea.

jksf replied:
Fixed in e337f0f

biern commented on test/signhash.ts:
```diff
@@ -142,4 +142,66 @@ contract('SignHash', accounts => {
+      await assertThrowsInvalidOpcode(async () => {
+        await instance.removeProof(method, { from: signer });
+      });
+    });
```
There should also be a test for removing a proof of another signer.

jksf replied:
Not possible. The proof is always related to `msg.sender`. It would be the same test as `should throw when proof does not exist`.

jksf replied:
Ok, I can add a test or two

jksf replied:
Fixed in 094e9ec

biern commented:
I haven't thought of using DNS verification, you're right, domain is
ambiguous then :+1:

2017-10-04 20:57 GMT+02:00 Jakub Stefanski <notifications@github.com>:

> *@jstefanski* commented on this pull request.
> ------------------------------
>
> In test/signhash.ts
> <https://github.com/SignHash/signhash-contracts/pull/7#discussion_r142762072>
> :
>
> > @@ -142,4 +142,66 @@ contract('SignHash', accounts => {
>        });
>      });
>    });
> +
> +  describe('#removeProof', () => {
> +    it('should remove proof method', async () => {
> +      const signer = accounts[0];
> +      const method = 'http';
>
> https == HTTP over SSL. When accessing a static file on a server we are
> using HTTP protocol in the first place. I thought about using SSL
> certificate as identity proof but I have no idea yet how to integrate it
> with our identity verification mechanism.
>
> I would avoid using domain method instead of http as it would collide
> with dns custom records verification which is another way to proof the
> domain.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SignHash/signhash-contracts/pull/7#discussion_r142762072>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACjhm6ftj14JlQefc6pyJcYxfHpfOpJks5so9UKgaJpZM4Pp_WK>
> .
>



-- 
Pozdrawiam
  Marcin Biernat

Commit:
Clean up tests

Commit:
Test removing proof by other account

jksf commented:
@biern Thanks for comments. I think I fixed all issues.
I also removed `from` property from all transactions where the default account was used to make it more concise. I think it is more clear now, but one must know that `accounts[0]` is used by default by `web3js`.
